Задание 1: юнит-тесты

RealMadSquirrel opened the pull request:
- Покрыты тестами классы Bun, Burger, Ingredient, Database.
 - Сформирован отчет о покрытии в файле report.html.
 - Добавлены библиотеки pytest, pytest-cov.

Commit:
Задание 1: юнит-тесты

 - Покрыты тестами классы Bun, Burger, Ingredient,Database.
 - Сформирован отчет о покрытии в файле report.html

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,34 @@
+
+@pytest.fixture
+def burger():
+    return Burger()
```
Нужно исправить: фикстуры не занимаются прокидыванием данных в тест, они выполняют логику предусловий\постусловий. Простые объекты создаются в тесте

RealMadSquirrel replied:
Убрала

AlexKitIt commented on ingredient_types.py:
Нужно исправить: файлы проекта менять нельзя. Тестовые данные к данным программы добавляться не могут, они хранятся в data

RealMadSquirrel replied:
Поменяла

AlexKitIt commented on tests/test_bun.py:
```diff
@@ -0,0 +1,30 @@
+    def test_get_type_of_price_of_bun_true(self):
+        assert type(self.bun.get_price()) == float
+
+    def test_get_name_of_bun_true(self, mock_bun):
+        bun = Bun(mock_bun.get_name.return_value, mock_bun.get_price.return_value)
+        assert bun.get_name() == ingredient_types.BUN_NAME
+
+    def test_get_price_of_bun_true(self, mock_bun):
+        bun = Bun(mock_bun.get_name.return_value, mock_bun.get_price.return_value)
+        assert bun.get_price() == ingredient_types.BUN_PRICE
```
Зачем нам мок, если выше мы создаем объект Bun?

RealMadSquirrel replied:
Убрала

AlexKitIt commented on tests/test_ingredient.py:
```diff
@@ -0,0 +1,29 @@
+                            ingredient_types.INGREDIENT_PRICE)
+
+    def test_type_of_ingredient_true(self):
+        assert self.ingredient.type == ingredient_types.INGREDIENT_TYPE_SAUCE
```
Нужно исправить: не проверен метод get_type()

RealMadSquirrel replied:
Добавила

Commit:
Задание 1: юнит-тесты Правка №1

 - Убрала создание бургера burger() из фикстуры
 - Переместила тестовые данные в файл data, убрала из ingredient_types.py
 - добавлен метод  get_type() для ingredient
 - добавлен новый отчет report_2.html
